nginx/etc/nginx/includes/proxy-settings.conf

```diff
@@ -4,5 +4,6 @@
 
 proxy_set_header Host $http_host;
 proxy_set_header X-Forwarded-For $remote_addr;
+proxy_connect_timeout 5s;
 proxy_read_timeout 20s;
```

Comment:
The analysis above did not take into consideration that the time the request spends in Nginx land is only a sliver of the total time (when ALB generates response - when ALB receives the request). I updated the query to use `target_processing_time` which should increase the precision of the analysis.

I used variations of this query to determine probability of whether a request, after **N seconds**, over the past **6 months**, will be successful:

```sql
SELECT COUNT(*) FILTER (WHERE target_processing_time BETWEEN 5 AND 15) AS between_5_and_15, 
       COUNT(*) FILTER (WHERE target_processing_time > 15) AS more_than_15
FROM "raster_foundry"."production_backsplash_alb_logs"
WHERE month(from_iso8601_timestamp(time)) >= 3
        AND year(from_iso8601_timestamp(time)) = 2019
        AND target_status_code = '200'
UNION
SELECT COUNT(*) FILTER (WHERE target_processing_time BETWEEN 5 AND 15) AS between_5_and_15, 
       COUNT(*) FILTER (WHERE target_processing_time > 15) AS more_than_15
FROM "raster_foundry"."production_backsplash_alb_logs"
WHERE month(from_iso8601_timestamp(time)) >= 3
        AND year(from_iso8601_timestamp(time)) = 2019
```

||after 1 second| after 5 seconds | after 10 seconds | after 15 seconds |
|---|---|-----------------|------------------|------------------|
|probability of success|97%| 34%             | 11%              | 0.8%               |
|percent of successful requests that reached 1 second| |17%| 4%|0.2%|

Since successful requests that reach **5 seconds** are a minority, I think you could argue that would work. I'm leaning towards **10 seconds**. That would move us to a better place and have a decent sized buffer.

Reply:
Nice investigation - this helps clarify the options a bunch. I think 10 seconds is probably a good first pass.